scss/_settings_colors.scss

```diff
@@ -31,6 +31,9 @@ $grid-border-style: 1px dotted $color-mid-light !default;
 
 // for _patterns_navigation.scss
 $color-navigation-background: $color-x-light !default;
+$color-navigation-border: if(lightness($color-navigation-background) > 50, $color-mid-light, transparent) !default;
+$color-navigation-text: if(lightness($color-navigation-background) > 70, $color-dark, $color-light) !default;
+
 $color-navigation-selected: #757575 !default;
```

Comment:
• I'm a bit concerned that doing this automatically based on lightness doesn't guarantee a good result. If the point is to provide a reasonable default for uses not covered by the default component, maybe no border is a better default? E.g. the ubuntu nav (and maas.io, jaas.ai, snapcraft.io all have to override this because they don't need a border). 
• Is there a reason why these are moved (including the conditionals) to a settings file? Shouldn't settings be for more universal things? Like optimal border on white background, which then gets used by tabs, hrs, menus, anything that needs to match. Otherwise all of these could clutter the settings file, when in fact they all refer to the same grey value.